src/main/java/org/dependencytrack/tasks/OsvDownloadTask.java

```diff
@@ -105,32 +108,41 @@ public OsvDownloadTask() {
-                    String url = this.osvBaseUrl + URLEncoder.encode(ecosystem, StandardCharsets.UTF_8).replace("+", "%20")
-                            + "/all.zip";
-                    HttpUriRequest request = new HttpGet(url);
-                    try (var ignoredMdcOsvEcosystem = MDC.putCloseable("osvEcosystem", ecosystem);
-                         final CloseableHttpResponse response = HttpClientPool.getClient().execute(request)) {
-                        final StatusLine status = response.getStatusLine();
-                        if (status.getStatusCode() == HttpStatus.SC_OK) {
-                            try (InputStream in = response.getEntity().getContent();
-                                 ZipInputStream zipInput = new ZipInputStream(in)) {
-                                unzipFolder(zipInput);
-                            }
-                        } else {
-                            LOGGER.error("Download failed : " + status.getStatusCode() + ": " + status.getReasonPhrase());
-                        }
-                    } catch (Exception ex) {
-                        LOGGER.error("Exception while executing Http client request", ex);
+        if (this.ecosystems == null || this.ecosystems.isEmpty()) {
+            LOGGER.info("Google OSV mirroring is disabled. No ecosystem selected.");
+            return;
+        }
+
+        for (final var ecosystem : this.ecosystems) {
+            LOGGER.info("Updating datasource with Google OSV advisories for ecosystem " + ecosystem);
+            final var url = this.osvBaseUrl + URLEncoder.encode(ecosystem, StandardCharsets.UTF_8).replace("+", "%20") + "/all.zip";
+            final var request = new HttpGet(url);
+            try (final var ignoredMdcOsvEcosystem = MDC.putCloseable("osvEcosystem", ecosystem); final var response = HttpClientPool.getClient().execute(request)) {
+                final var status = response.getStatusLine();
+                if (status.getStatusCode() != HttpStatus.SC_OK) {
+                    LOGGER.error("Download failed : " + status.getStatusCode() + ": " + status.getReasonPhrase());
+                    continue;
+                }
+
+                final var tempFile = File.createTempFile("google-osv-download", ".zip");
+                try {
+                    try (final var out = new FileOutputStream(tempFile); final var in = response.getEntity().getContent()) {
+                        out.write(in.readAllBytes());
+                    }
+                    try (final var in = new FileInputStream(tempFile); final var zipInput = new ZipInputStream(in)) {
+                        unzipFolder(zipInput);
+                    }
+                } finally {
+                    if (!tempFile.delete()) {
+                        LOGGER.warn("Failed to delete temporary file: " + tempFile.getAbsolutePath());
                     }
                 }
```

Comment:
Consider replacing this manual call to `delete` with passing [`StandardOpenOption.DELETE_ON_EXIT`](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/nio/file/StandardOpenOption.html#DELETE_ON_CLOSE) to [`Files.newInputStream`](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/nio/file/Files.html#newInputStream(java.nio.file.Path,java.nio.file.OpenOption...)).